src/status_im/communities/core.cljs

```diff
@@ -87,23 +87,27 @@
   (:communities/community-id-input db))
 
 (defn- handle-my-request
-  [db {:keys [community-id state] :as request}]
-  {:db (if (= constants/community-request-to-join-state-pending state)
-         (assoc-in db [:communities/my-pending-requests-to-join community-id] request)
-         (update-in db [:communities/my-pending-requests-to-join] dissoc community-id))})
+  [db {:keys [community-id state deleted] :as request}]
+  (if (and (= constants/community-request-to-join-state-pending state) (false? deleted))
```

Comment:
I think it's more robust in general in Clojure to take advantage of *falsey* conditionals. If we use `(not deleted)` it correctly evaluates to true if `deleted` is nil, but `false?` only evaluates to true if `deleted` is exactly equal to `false`. Since many things can be nil in Clojure, using `false?` can be a recipe for a subtle bug in the future.

Although less important, a similar logic applies to `(if (true? deleted) ...)`, we can use `(if (some? deleted) ...)` or just `(if deleted)`.

Reply:
I agree. I update it. 👍 Thanks for the awesome explanation, Icaro. 😄